core/logger/zap.go

```diff
@@ -16,14 +17,16 @@ import (
 var _ zapcore.Core = &AtomicCore{}
 
 type AtomicCore struct {
-	mu       sync.RWMutex
-	core     zapcore.Core
-	children []weak.Pointer[withCore]
+	mu              sync.RWMutex
+	core            zapcore.Core
+	children        []weak.Pointer[withCore]
+	lastCleaned     time.Time
+	cleanupInterval time.Duration
 }
 
-// NewAtomicCore creates a new AtomicCore initialized with a noop core
-func NewAtomicCore() *AtomicCore {
-	return &AtomicCore{core: zapcore.NewNopCore()}
+// NewAtomicCore creates a new AtomicCore initialized with a noop core.
+func NewAtomicCore(cleanupInterval time.Duration) *AtomicCore {
```

Comment:
Might want to avoid modifying this exported API and leave it as an implementation detail